### DISCUSSION TITLE
Link to devices in 'Support New Device' end up in 404.

### DISCUSSION THREAD

**Discussion by dbenyakar** [Koenkk/zigbee2mqtt#17930] [Devices]
Im new to adding new device and Im trying to add new device which seems to not being currently supported.
Started following the instructions which refer to existing examples.
However the link to [devices ](https://github.com/Koenkk/zigbee-herdsman-converters/blob/master/devices) (at least for a noob like me), does not work. Either the page moved or been deleted.
Where can I find these examples? 
Same thing for [Converters ](https://github.com/Koenkk/zigbee-herdsman-converters/blob/master/converters/fromZigbee.js), again either the page moved or been deleted.

**Comment by sjorge** [comment]
The zhc repo is ondergoing a js -> ts migration atm and stuff is moving around a bit.

Devices are currently here https://github.com/Koenkk/zigbee-herdsman-converters/tree/master/src/devices
Converters are here https://github.com/Koenkk/zigbee-herdsman-converters/tree/master/src/converters

Not sure how accurate these instructions still are, you can't just edit the files in place anymore. The easiest way to do development i found is to link in (with npm link) the zhc package, the run `npm run build` before (re)starting z2m.

It's probably best to wait until after the migration if it's your first time trying to add a device.

**Reply by dbenyakar** [reply]
I see. OK thanks for this.
To be on the safe side I ended up opening a ticket for new device support. I didn't want to mess things up.
